cli/index.ts

```diff
@@ -173,10 +182,19 @@ const builtYargs = createYargsCli({
             }
           }
         };
-        fs.writeFileSync(
-          path.resolve(argv["project-dir"], CREDENTIALS_FILENAME),
-          prettyJsonStringify(credentials())
-        );
+        const credentials = credentialsFn();
+        if (argv["test-connection"]) {
+          try {
+            await utils.testCredentials(credentials, argv.warehouse);
+            writeStdOut(commandOutput("Warehouse test query completed successfully."));
+          } catch (e) {
+            throw new Error(`Warehouse test query failed: ${e.stack || e.message}`);
+          }
+        }
+        const filePath = path.resolve(argv["project-dir"], CREDENTIALS_FILENAME);
+        fs.writeFileSync(filePath, prettyJsonStringify(credentials));
+        writeStdOut(commandOutput("Credentials file written:"));
```

Comment:
Want to add a comment saying to change settings that you can edit this file directly?

Reply:
Done